tools/android/wireguard-build.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+# Copyright (c) 2025 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+# This script is used to build the WireGuard tunnel library.
+# This should be ran from the Brave root. aka `root/src/brave/`.
+pushd .
+
+# uses JAVA and SDK version from Chromium checkout.
+# this syntax w/ realpath is expanding relative path to absolute path.
+export JAVA_HOME=`realpath ../third_party/jdk/current`
+export PATH=$PATH:$JAVA_HOME/bin
+export ANDROID_HOME=`realpath ../third_party/android_sdk/public`
+export PATH="$PATH:$ANDROID_HOME/cmdline-tools/latest/bin/"
+echo "JAVA_HOME=$JAVA_HOME"
+echo "ANDROID_HOME=$ANDROID_HOME"
+
+BRAVE_DEPS_DIR=`realpath ./third_party/android_deps/libs/com_wireguard_android`
+echo "Changing directory to $BRAVE_DEPS_DIR"
+cd "$BRAVE_DEPS_DIR"
+pushd .
+
+echo "If you haven't accepted licenses yet this is interactive."
+sdkmanager --licenses
+
+WIREGUARD_TAG="1.0.20250531"
+REPO_URL="https://git.zx2c4.com/wireguard-android"
```

Comment:
this should be a conditional dep in DEPS, maybe `(checkout_wireguard and (checkout_android or checkout_linux))` guard where `checkout_wireguard` is added to `vars = { ...  }`
Also we should use the commit ref instead of tags because tags can be changed and potentially result in building different code than we expect